ystia/yorc
# GH 371 remove consul types duplicates

# Pull Request description

## Description of the change

Store TOSCA types coming with Yorc in a common path of Consul instead of duplicating them into each deployment.

### What I did


### How to verify it

1. Deploy some applications using a Yorc on develop.
2. Use `curl -s "http://127.0.0.1:8500/v1/kv/_yorc/deployments?recurse" | /snap/bin/jq '[.[] | (.Value + "" | @base64d ) + .Session + .Key | utf8bytelength + 32] | add '` to estimate the size of the deployments path in Consul (jq 1.6+ is required)
3. Then restart with this version of Yorc. 
4. Use  `curl -s "http://127.0.0.1:8500/v1/kv/_yorc/deployments?recurse" | /snap/bin/jq '[.[] | (.Value + "" | @base64d ) + .Session + .Key | utf8bytelength + 32] | add '` to check that the size has decreased.
5. An deploy applications and redeploy them to check that every thing is ok 


### Description for the changelog

* Reduce the volume of data stored in Consul by removing builtin types duplicates on deployments ([GH-371](https://github.com/ystia/yorc/issues/371))

## Applicable Issues

Fixes #371
